reference-service/src/main/java/com/transformuk/hee/tis/reference/service/api/FundingTypeResource.java

```diff
@@ -120,9 +120,20 @@ public ResponseEntity<FundingTypeDTO> updateFundingType(
     if (fundingTypeDTO.getId() == null) {
       return createFundingType(fundingTypeDTO);
     }
-    FundingType fundingType = fundingTypeMapper.fundingTypeDTOToFundingType(fundingTypeDTO);
-    fundingType = fundingTypeRepository.save(fundingType);
-    FundingTypeDTO result = fundingTypeMapper.fundingTypeToFundingTypeDTO(fundingType);
+
+    Optional<FundingType> optionalFundingType = fundingTypeRepository.findById(
+        fundingTypeDTO.getId());
+    if (optionalFundingType.isEmpty()) {
+      return ResponseEntity.badRequest().headers(HeaderUtil
+              .createFailureAlert(ENTITY_NAME, "id_not_found",
+                  "Cannot update FundingType. FundingType with given id not found"))
+          .body(null);
+    }
+    FundingType fundingTypeFromDb = optionalFundingType.get();
+    // map the changes and set into the entity from db
+    fundingTypeMapper.updateFundingTypeFromDto(fundingTypeDTO, fundingTypeFromDb);
+    fundingTypeFromDb = fundingTypeRepository.save(fundingTypeFromDb);
```

Comment:
The `load & patch' algorithm only prevents a uuid from changing.  That probably makes sense but feels like an assumption we should validate.  In particular if it's not valid to change that identifier, we ought to check it for consistency.